app/src/main/java/ch/epfl/skysync/viewmodel/FlightsViewModel.kt

```diff
@@ -133,12 +139,18 @@ class FlightsViewModel(
                   localDate = date!!,
                   timeslot = timeSlot!!,
                   onError = { onError(it) })
+
+          if (_flight.value?.balloon != null && firstFill) {
```

Comment:
Instead of checking if it is the first time it is refreshed, you should check if the date is the date of the flight, that way you can add the flight members as available users when the date match and not otherwise

Reply:
Good catch. I will change it